core/src/mindustry/world/Block.java

```diff
@@ -624,6 +624,18 @@ protected void initBuilding(){
 
         return out;
     }
+    
+    /** @return if all booster items are unlocked. If it doesn't have a booster, it outputs false */
+    public boolean boosterUnlocked(){
+        boolean unlocked = true;
+        for(ItemStack item : consumes.getItem().items){
+            if(!item.item.unlockedNow() || !consumes.getItem().isBoost()){
+                unlocked = false;
+                break;
+            }
+        }
+        return unlocked;
+    }
```

Comment:
Maybe change it like this?

```suggestion
    public boolean boosterUnlocked(){
        for(ItemStack item : consumes.getItem().items){
            if(!item.item.unlockedNow() || !consumes.getItem().isBoost()){
                return false;
            }
        }
        return true;
    }
```